torch/fx/symbolic_trace.py

```diff
@@ -106,6 +106,44 @@ def create_arg(self, a: Any) -> Argument:
                 if a is p:
                     return self.graph.get_param(n)
             raise NameError('parameter is not a member of this module')
+        # Tensors do not have a reliable string repr() from which they can be
+        # constructed (and we probably don't want to rely on that, either), so
+        # for any constant Tensor values we encounter, first search for if they
+        # are an attribute of some module in the module hierarchy. If so, emit
+        # a get_param to retrieve that tensor. Otherwise, we'll store away the
+        # tensor value into a special attribute on the Module s.t. we can
+        # retrieve it with a get_param.
+        if isinstance(a, torch.Tensor):
+            # TODO: slow
+            def search_for_tensor(m : torch.nn.Module, qualname_atoms : List[str]) -> Optional[str]:
+                """
+                Search for a tensor value in the module's attributes. If it's
+                found, return the qualified name of that attribute, given the
+                previous `qualname_atoms`. If it's not found, recurse down into
+                child submodules. If it's not found there, return None
+                """
+                for n, p in m.__dict__.items():
+                    if isinstance(p, torch.Tensor) and a is p:
+                        return '.'.join(qualname_atoms + [n])
+                for n, c in m.named_children():
+                    maybe_result : Optional[str] = search_for_tensor(c, qualname_atoms + [n])
+                    if maybe_result:
+                        return maybe_result
+                return None
+            # Retrieve the qualname for an existing Tensor attribute
+            qualname : Optional[str] = search_for_tensor(self.root, [])
+
+            # Tensor was not found in the Module hierarchy, stow it away in a
+            # special attribute and set the qualname to refer to that
+            if not qualname:
+                i = 0
+                while True:
+                    qualname = f'__tensor_constant{i}'
+                    if not hasattr(self.root, qualname):
```

Comment:
I don't like how this is being dumped on `root`. Ideally it would be somewhere else, like on the produced GraphModule. @zdevito @suo any suggestions?

Reply:
it should be naturally addressed if you eliminate `root` and instead copy submodules (while filtering only the ones that occur in `get_param`). We talked about it all together but I'm not sure whether anyone is working on it (@zdevito ?)